### PR TITLE
Update SAM libraries

### LINKED ISSUE #440: update sam library files
We've fallen a few years behind the SAM releases, so I think it's time to update the module and inverter CEC files.

### PR DESCRIPTION
**Pull request by cwhanse**
pvlib python pull request guidelines
====================================

Thank you for your contribution to pvlib python!

You may submit a pull request with your code at any stage of completion, however, before the code can be merged the following items must be addressed:

 - [x] Closes #440 
 - [x] Fully tested. Added and/or modified tests to ensure correct behavior for all reasonable inputs. Tests must pass on the TravisCI and Appveyor testing services.
 - [x] Code quality and style is sufficient. Passes ``git diff upstream/master -u -- "*.py" | flake8 --diff`` and/or landscape.io linting service.
 - [x] New code is fully documented. Includes sphinx/numpydoc compliant docstrings and comments in the code where necessary.
 - [x] Updates entries to `docs/sphinx/source/api.rst` for API changes.
 - [x] Adds description and name entries in the appropriate `docs/sphinx/source/whatsnew` file for all changes.

Please don't hesitate to ask for help if you're unsure of how to accomplish any of the above. You may delete all of these instructions except for the list above.

Brief description of the problem and proposed solution (if not already fully described in the issue linked to above):

### CODE REVIEW

**Comment by wholmgren** [comment]
YES!!! 

This mostly looks good to me. It looks like you accidentally changed the line for looking up the adr inverter file instead of the cec inverter file.

We should add a comment to the whats new file. 

I think we should remove the old files, too, but that is up for debate.

```
git rm pvlib/data/sam-library-cec-inverters-2015-6-30.csv
git rm pvlib/data/sam-library-cec-modules-2015-6-30.csv
```

**Comment by cwhanse** [comment]
We can remove the old files.  I checked that the new files are adding entries, there aren't any old ones that will disappear.  I checked that the pvlib function reads the new files.



And, it figures that once the commit got through it would have a mistake in it.





________________________________

**Comment by cwhanse** [comment]
Fixed the inverter file lines

**Comment by wholmgren** [comment]
It appears that the "Example Module" parameters are slightly different, so that explains a few of the failing tests. 

I think the others are failing because some of the fields are now read in as strings instead of as numbers. Maybe there is a slight change to the format or a new name that prevents pandas from assuming that the fields are numeric?

**Comment by wholmgren** [comment]
I think the problem is line 2508 in the inverters file:

```
LeadSolar Energy: LS600T [208V] 208V [CEC 2018],208,500,75790.4,36,-nan(ind),1.22899e-07,-0.0251886,-nan(ind),0.0798754,1,45,2105.29,27,45
```

**Comment by cwhanse** [comment]
Glad you tracked that line down.  I got as far are 'there's something different in format between these files'

**Comment by wholmgren** [comment]
@cwhanse up to you on if we should hold off for the SAM team to update the files. In any case, you'll need to ``git rm`` the 2015 files, then commit the change.

**Comment by cwhanse** [comment]
I talked with NREL, we should edit the SAM libraries and move ahead with our merge.  The alternative is to wait for the next SAM release, since that is when the SAM libraries are updated.

**Comment by thunderfish24** [comment]
@cwhanse Thanks for working thru the issues with the new version. FWIW, while I was "scraping" the existing cec module dataset for single cells (i.e., Ns=1), my scraper code caught a few entries with missing/bad data.

**Comment by wholmgren** [comment]
@cwhanse and @thunderfish24: do you want to remove those modules too or go ahead and merge? They don't seem to cause a problem for the import function, so my vote is to move on.

**Comment by cwhanse** [comment]
I’d keep any module which has a full set of parameter values.  We don’t have a system to judge good from bad parameter sets, so we have no basis to remove a module. Even when a module has a questionable value (e.g. negative resistance, diode factor < 1) the calculation could still produce usable IV curves.

Sent from my iPhone

On Apr 6, 2018, at 10:17 AM, Will Holmgren <notifications@github.com<mailto:notifications@github.com>> wrote:


@cwhanse<https://github.com/cwhanse> and @thunderfish24<https://github.com/thunderfish24>: do you want to remove those modules too or go ahead and merge? They don't seem to cause a problem for the import function, so my vote is to move on.

**Comment by wholmgren** [comment]
The only outstanding issue is to add an entry to the 0.5.2.rst whatsnew file. I don't think there should be any problem with merge commits because git is set to union those files by default.

**Comment by cwhanse** [comment]
@mikofski thanks for that illustration.  I'm not opposed to adopting parts of pvfree into pvlib, as a replacement for the current method of reading csv files of model parameters.  But we'd want to agree that the change is needed and is an improvement.  I don't see that pvlib is ready for this change.  The problem is providing a traceable and stable input files.  The CEC databases contain module and inverter measurements, not model parameters.  The SAM libraries are output of processing the public CEC databases using tools that are now in the SAM SDK, but using those tools to produce the library files is currently somewhat ad hoc. Since we can't produce the model parameters natively in pvlib by processing the CEC databases, we rely on a particular SAM release to provide traceable and stable inputs. Because of this reliance on SAM, I think we're better using the files directly, than we would be with an additional layer of code between the files and pvlib.

I hope this makes sense.

**Comment by cwhanse** [comment]
Note added for whatsnew

**Comment by wholmgren** [comment]
Hm, I don't see the changes to whatsnew. Maybe missed a ``commit`` or ``push`` step?

**Comment by wholmgren** [comment]
There it is. I don't understand why there is a merge conflict. I tested it locally and it worked fine on my machine. Irritating.

Can you click on the resolve conflicts button and remove the three marked lines?

**Comment by wholmgren** [comment]
Great, thanks!

**Comment by cedricleroy** [comment]
I think https://github.com/pvlib/pvlib-python/commit/f892664b255dc382b9e87f3340ff768ef31ba0ee did overwrite a couple of previous things in `v0.5.2.rst` 

There is a "0" here: https://github.com/pvlib/pvlib-python/commit/f892664b255dc382b9e87f3340ff768ef31ba0ee#diff-2df8cbc501f75a90135963ace8a28f63R1

And the text here has been removed: https://github.com/pvlib/pvlib-python/commit/f892664b255dc382b9e87f3340ff768ef31ba0ee#diff-2df8cbc501f75a90135963ace8a28f63L15

File: docs/sphinx/source/whatsnew/v0.5.2.rst [hunk 1]
```diff
@@ -12,8 +12,10 @@ API Changes
 
 Enhancements
 ~~~~~~~~~~~~
-* Improve clearsky.lookup_linke_turbidity speed, changing .mat source
-  file to .h5 (:issue:`437`)
+* Updated libraries for CEC module parameters to SAM release 2017.9.5 
+(library dated 2017.6.30) and CEC inverter parameters to file posted to 
+www.github.com/NREL/SAM on 2018.3.18, with one entry removed due to a 
+missing parameter value. (:issue:'440')
```
Comment (mikofski) [inline docs/sphinx/source/whatsnew/v0.5.2.rst]: please change single quotes around #440 to backticks

**Pull request merged**